posydon/unit_tests/popsyn/test_norm_pop.py

```diff
@@ -0,0 +1,839 @@
+        assert np.allclose(result, 0.15)        
+
+class TestGetMeanMass:
+    def test_dummy_mean_mass(self):
+        # dummy PDF returns 1 for any input
+        dummy_pdf = lambda *args, **kwargs: 1
+        # parameters chosen for analytic integration:
+        # m1 in [1,2], q in [0.2, 0.8]
+        params = {
+            'primary_mass_min': 1,
+            'primary_mass_max': 2,
+            'q_min': 0.2,
+            'q_max': 0.8
+        }
+        # Binary integration:
+        # I_bin = ∫[m=1 to 2] m * (∫[q=0.2 to 0.8] (1+q)dq) dm
+        # ∫[q=0.2 to 0.8] (1+q)dq = [(q + 0.5*q^2)]_0.2^0.8 
+        # = (0.8+0.32) - (0.2+0.02) = 1.12-0.22 = 0.9
+        # I_bin = 0.9 * ∫[m=1 to 2] m dm = 0.9*( (2^2-1^2)/2 ) = 0.9*1.5 = 1.35
+        # Single star integration:
+        # I_single = ∫[m=1 to 2] m dm = 1.5
+        # Total expected mean mass = 1.35+1.5 = 2.85
+        result = norm_pop.get_mean_mass(dummy_pdf, params)
+        assert np.allclose(result, 2.85, rtol=1e-2)
+
+    def test_dummy_mean_mass_without_q_bounds(self):
+        # dummy PDF returns 1 for any input
+        dummy_pdf = lambda *args, **kwargs: 1
+        # parameters without q_min and q_max, but with 
+        #  secondary_mass_min and secondary_mass_max:
+        # Let secondary_mass_min = 0.5, secondary_mass_max = 1.5.
+        # Default q_min = max(secondary_mass_min/primary_mass_min, 0) 
+        # = max(0.5/1,0)=0.5
+        # Default q_max = min(secondary_mass_max/primary_mass_max, 1)
+        # = min(1.5/2,1)=0.75
+        # Binary integration:
+        # Inner integration for each m: ∫[q=0.5 to 0.75] (1+q)dq
+        # = [(q + 0.5*q^2)] from 0.5 to 0.75 = (0.75+0.28125) - (0.5+0.125) 
+        # = 1.03125 - 0.625 = 0.40625
+        # Outer integration: ∫[m=1 to 2] m*0.40625 dm = 0.40625*( (2^2-1^2)/2 ) 
+        # = 0.40625*1.5 = 0.609375
+        # Single star integration: ∫[m=1 to 2] m dm = 1.5
+        # Total expected mean mass = 0.609375 + 1.5 = 2.109375
+        params = {
+            'primary_mass_min': 1,
+            'primary_mass_max': 2,
+            'secondary_mass_min': 0.5,
+            'secondary_mass_max': 1.5
+        }
+        result = norm_pop.get_mean_mass(dummy_pdf, params)
+        assert np.allclose(result, 2.109375, rtol=1e-2)
+
+####
+# Test the reweighting function
+#### 
+
+
+@fixture
+def base_simulation_kwargs():
+    simulation_params = {'number_of_binaries': 100000,
+          'primary_mass_scheme':'Kroupa2001',
+          'binary_fraction_const':1.0,
+          'binary_fraction_scheme':'const',
+          'primary_mass_min': 7,
+          'primary_mass_max': 140,
+          'secondary_mass_scheme':'flat_mass_ratio',
+          'secondary_mass_min':0.35,
+          'secondary_mass_max':140,
+          'orbital_scheme':'period',
+          'orbital_period_scheme':'Sana+12_period_extended',
+          'orbital_period_min':0.35,
+          'orbital_period_max':6e3,
+          'eccentricity_scheme':'zero',}
```

Comment:
Just a bit of style:
```suggestion
    simulation_params = {'number_of_binaries': 100000,
                         'primary_mass_scheme': 'Kroupa2001',
                         'primary_mass_min': 7,
                         'primary_mass_max': 140,
                         'binary_fraction_scheme': 'const',
                         'binary_fraction_const': 1.0,
                         'secondary_mass_scheme': 'flat_mass_ratio',
                         'secondary_mass_min': 0.35,
                         'secondary_mass_max': 140,
                         'orbital_scheme': 'period',
                         'orbital_period_scheme': 'Sana+12_period_extended',
                         'orbital_period_min': 0.35,
                         'orbital_period_max': 6e3,
                         'eccentricity_scheme': 'zero',}
```